thunder/benchmarks/benchmark_litgpt.py

```diff
@@ -396,6 +408,24 @@ def setup_compile(self, model):
                 executors.insert(0, transformer_engine_ex)
 
             if "dynamo" in self.compile:
+                if self.distributed_mode == "fsdp2":
+                    print("Resetting cache size for when fsdp2 and using thunder as backend torch.compile")
+                    import torch._dynamo.config as dynamo_config
+
+                    dynamo_config.cache_size_limit = 64
+                if "transformerengine" in self.compile:
+                    # [rank0]:   File "/opt/pytorch/lightning-thunder/thunder/executors/transformer_engineex.py", line 410, in _te_functional_linear_backward_impl
+                    # [rank0]:     grads = _Linear.backward(ctx, g)
+                    # [rank0]:   File "/usr/local/lib/python3.10/dist-packages/transformer_engine/pytorch/module/linear.py", line 449, in backward
+                    # [rank0]:     weight_fp8.transpose_2d(),
+                    # [rank0]:   File "/usr/local/lib/python3.10/dist-packages/transformer_engine/pytorch/float8_tensor.py", line 625, in transpose_2d
+                    # [rank0]:     if self._transpose is None:
+                    # [rank0]:   File "/usr/local/lib/python3.10/dist-packages/transformer_engine/pytorch/float8_tensor.py", line 39, in get_func
+                    # [rank0]:     return self._fp8_attrs[name]
+                    # [rank0]: AttributeError: 'Float8Tensor' object has no attribute '_fp8_attrs'
+                    raise ValueError(
+                        "TransformerEngine executor cannot be used as an executor of Thunder when Thunder is used as torch.compile backend"
+                    )
```

Comment:
@kshitij12345, if you have time could you please investigate what's happening here?

Reply:
ah this check should be in the if stmt of fsdp2...